task_sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -163,6 +174,114 @@ def __eq__(self, other: object) -> bool:
         return True
 
 
+@attrs.define
+class AssetAliasEvent:
+    """Representation of asset event to be triggered by an asset alias."""
+
+    source_alias_name: str
+    dest_asset_key: AssetUniqueKey
+    extra: dict[str, Any]
+
+
+@attrs.define
+class OutletEventAccessor:
+    """Wrapper to access an outlet asset event in template."""
+
+    key: BaseAssetUniqueKey
+    extra: dict[str, Any] = attrs.Factory(dict)
+    asset_alias_events: list[AssetAliasEvent] = attrs.field(factory=list)
+
+    def add(self, asset: Asset, extra: dict[str, Any] | None = None) -> None:
+        """Add an AssetEvent to an existing Asset."""
+        if not isinstance(self.key, AssetAliasUniqueKey):
+            return
+
+        asset_alias_name = self.key.name
+        event = AssetAliasEvent(
+            source_alias_name=asset_alias_name,
+            dest_asset_key=AssetUniqueKey.from_asset(asset),
+            extra=extra or {},
+        )
+        self.asset_alias_events.append(event)
+
+
+def _get_asset(name: str | None = None, uri: str | None = None) -> Asset:
+    from airflow.sdk.definitions.asset import Asset
+    from airflow.sdk.execution_time.comms import ErrorResponse, GetAssetByName, GetAssetByUri
+    from airflow.sdk.execution_time.task_runner import SUPERVISOR_COMMS
+
+    if name:
+        SUPERVISOR_COMMS.send_request(log=log, msg=GetAssetByName(name=name))
+    elif uri:
+        SUPERVISOR_COMMS.send_request(log=log, msg=GetAssetByUri(uri=uri))
+    else:
+        raise ValueError("Either name or uri must be provided")
+
+    msg = SUPERVISOR_COMMS.get_message()
+    if isinstance(msg, ErrorResponse):
+        raise AirflowRuntimeError(msg)
+
+    if TYPE_CHECKING:
+        assert isinstance(msg, AssetResult)
+    return Asset(**msg.model_dump(exclude={"type"}))
+
+
+class OutletEventAccessors(Mapping[Union[Asset, AssetAlias], OutletEventAccessor]):
```

Comment:
I'm not too familiar with this context key, but aren'te Outlets defined on the task, i.e. we shouldn't need to go to the server for this?

Reply:
It is because now we can reference an Asset without requiring original `Asset` object using `Asset.ref` / `AssetRef` example: `Asset.ref(name="inlet_asset_1")` for [AIP-75](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-75+New+Asset-Centric+Syntax):
Was added in https://github.com/apache/airflow/pull/41325 & enhanced in https://github.com/apache/airflow/pull/45028